layouts/contact/list.html

```diff
@@ -3,55 +3,159 @@
+        <div class="col-sm-10">
+          <input type="text" class="form-control" id="company" name="company" tabindex="-1" autocomplete="off">
+        </div>
       </div>
-    </div>
-    <div class="g-recaptcha" data-sitekey="6LdPd50fAAAAAJ_dVMXFJjzOTh7XZeAI-ej82GPv"></div>
-    <label id="g-recaptcha" for="url">
-    <input type="url" id="originUrl" name="originUrl" value="" hidden aria-labelledby="g-recaptcha" />
-    <label id="general_contact" for="url">
-    <input type='hidden' name='origin' value='general_contact' hidden aria-labelledby="general_contact">
-    <button type="submit" class="btn btn-aqua mt-2">Send</button>
-  </form>
+      
+      <!-- Message field -->
+      <div class="row mb-3">
+        <label for="message" class="col-sm-2 col-form-label fw-bold">Message:*</label>
+        <div class="col-sm-10">
+          <textarea class="form-control" id="message" name="message" rows="5" required></textarea>
+        </div>
+      </div>
+      
+      <!-- reCAPTCHA - commented out to test anti-spam -->
+      <!-- <div class="g-recaptcha" data-sitekey="6LdPd50fAAAAAJ_dVMXFJjzOTh7XZeAI-ej82GPv"></div> -->
+      <!-- <label id="g-recaptcha" for="url"> -->
+      
+      <button type="button" class="btn btn-aqua mt-2" id="submit-btn" onclick="submitContactForm()">Send</button>
+    </form>
+  </div>
 </div>
 
 <script>
-  document.getElementById('originUrl').value = location.href;
+  // Initialise session tracking for anti-spam
+  var sessionStartTime = Date.now();
+  var sessionId = generateSessionId('contact');
 
-  var form = document.querySelector('.needs-validation');
-  form.addEventListener('submit', function (event) {
-    if (form.checkValidity() === false) {
-      event.preventDefault();
-      alert("Please fill out all required fields.");
-      event.stopPropagation();
-      form.classList.add('was-validated')
+  // Prevent form submission if user presses Enter
+  preventEnterSubmission('contact-form');
+
+  function submitContactForm() {
+    // Validate form-specific fields
+    if (!validateContactForm()) {
+      return;
     }
+
+    // Validate common submission criteria
+    if (!validateCommonSubmission(sessionStartTime)) {
+      showContactError('Form submission failed validation. Please try again.');
+      return;
+    }
+
+    // reCAPTCHA validation - commented out to test anti-spam
+    /*
     var recaptcha = $("#g-recaptcha-response").val();
     if (recaptcha === "") {
-      event.preventDefault();
-      alert("Please tick 'I'm not a robot' above the 'Submit the form' button.");
+      showContactError("Please tick 'I'm not a robot' above the 'Submit' button.");
+      return;
     }
-  });
+    */
+
+    // Show loading state
+    setButtonLoadingState('submit-btn', true, 'Sending...');
+
+    // Hide any previous messages
+    hideFormMessage('success-message');
+    hideFormMessage('error-message');
+
+    // Collect form-specific data
+    var formData = new FormData();
```

Comment:
Any reason for not using the following as you have in the other files ?

```
var formData = new FormData(document.getElementById('contact-form'))
```

Reply:
None at all :sweat_smile: